offer-manager-domain/src/main/groovy/life/qbic/business/customers/create/CreateCustomer.groovy

```diff
@@ -15,17 +16,24 @@ class CreateCustomer implements CreateCustomerInput {
 
   private CreateCustomerDataSource dataSource
   private CreateCustomerOutput output
+  private UpdateCustomer updateCustomer
 
 
   CreateCustomer(CreateCustomerOutput output, CreateCustomerDataSource dataSource){
     this.output = output
     this.dataSource = dataSource
+    this.updateCustomer = new UpdateCustomer(output,dataSource)
   }
 
   @Override
   void createCustomer(Customer customer) {
     try {
-      dataSource.addCustomer(customer)
+      int customerId = dataSource.findCustomer(customer).get()
+      if(customerId){
+        updateCustomer.updateCustomer(customerId.toString(),customer)
+      }else{
+        dataSource.addCustomer(customer)
+      }
```

Comment:
I like this solution.
Now that everything is handled on this level, would it make sense to use the Integer version of customerId everywhere? We could get rid of the String parsing etc.

Reply:
Nice catch, I also thought so and currently removing the string :)